src/main/java/teammates/ui/controller/InstructorCourseRemindAction.java

```diff
@@ -105,54 +104,31 @@ public ActionResult execute() throws EntityDoesNotExistException {
         StringBuilder statusToAdmin = new StringBuilder(200);
         statusToAdmin.append("Registration Key sent to the following users in Course <span class=\"bold\">[")
                      .append(courseId)
                      .append("]</span>:<br>");
         
-        Iterator<Entry<String, JoinEmailData>> extractedEmailIterator =
-                extractEmailDataForLogging(emailsSent).entrySet().iterator();
+        Set<Entry<String, JoinEmailData>> entries = emailDataMap.entrySet();
         
-        while (extractedEmailIterator.hasNext()) {
-            Entry<String, JoinEmailData> extractedEmail = extractedEmailIterator.next();
+        for (Entry<String, JoinEmailData> entry : entries) {
             
-            String userEmail = extractedEmail.getKey();
-            JoinEmailData joinEmailData = extractedEmail.getValue();
+            String userEmail = entry.getKey();
+            JoinEmailData joinEmailData = entry.getValue();
             
-            statusToAdmin.append(joinEmailData.userName).append("<span class=\"bold\"> (").append(userEmail)
-                         .append(")</span>.<br>").append(joinEmailData.regKey).append("<br>");
+            statusToAdmin.append(joinEmailData.userName)
+                         .append("<span class=\"bold\"> (").append(userEmail).append(")</span>.<br>")
+                         .append(joinEmailData.regKey).append("<br>");
         }
         
         return statusToAdmin.toString();
     }
 
-    private Map<String, JoinEmailData> extractEmailDataForLogging(List<EmailWrapper> emails) {
-        Map<String, JoinEmailData> logData = new TreeMap<String, JoinEmailData>();
-        
-        for (EmailWrapper email : emails) {
-            String recipient = email.getRecipient();
-            String userName = extractUserName(email.getContent());
-            String regKey = extractRegistrationKey(email.getContent());
-            logData.put(recipient, new JoinEmailData(userName, regKey));
-        }
-        
-        return logData;
-    }
-    
-    private String extractUserName(String emailContent) {
-        int startIndex = emailContent.indexOf("Hello ") + "Hello ".length();
-        int endIndex = emailContent.indexOf(',');
-        return SanitizationHelper.desanitizeFromHtml(emailContent.substring(startIndex, endIndex));
-    }
-    
-    private String extractRegistrationKey(String emailContent) {
-        int startIndex = emailContent.indexOf("key=") + "key=".length();
-        int endIndex = emailContent.indexOf("\">http://");
-        if (endIndex < 0) {
-            endIndex = emailContent.indexOf("\">https://");
-        }
-        return emailContent.substring(startIndex, endIndex);
+    private String extractStudentRegistrationKey(StudentAttributes student) {
+        String joinLink = Config.getAppUrl(student.getRegistrationUrl()).toAbsoluteString();
+        int startIndex = joinLink.indexOf("key=") + "key=".length();
```

Comment:
This could be changed to use a constant from the `Const` class instead of keeping the old magic string.